src/transformers/models/chinese_clip/processing_chinese_clip.py

```diff
@@ -0,0 +1,115 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Image/Text processor class for Chinese-CLIP
+"""
+from ...processing_utils import ProcessorMixin
+from ...tokenization_utils_base import BatchEncoding
+
+
+class ChineseCLIPProcessor(ProcessorMixin):
+    r"""
+    Constructs a Chinese-CLIP processor which wraps a Chinese-CLIP feature extractor and a Chinese-CLIP tokenizer into
+    a single processor.
+
+    [`ChineseCLIPProcessor`] offers all the functionalities of [`ChineseCLIPFeatureExtractor`] and
+    [`BertTokenizerFast`]. See the [`~ChineseCLIPProcessor.__call__`] and [`~ChineseCLIPProcessor.decode`] for more
+    information.
+
+    Args:
+        feature_extractor ([`ChineseCLIPFeatureExtractor`]):
+            The feature extractor is a required input.
+        tokenizer ([`BertTokenizerFast`]):
+            The tokenizer is a required input.
+    """
+    feature_extractor_class = "ChineseCLIPFeatureExtractor"
```

Comment:
@amyeroberts For new models, could we (and do we want to) use `image_processor_class`, and more generally, avoid using the name feature_extractor?

Reply:
No, not yet. The feature extractors haven't officially been deprecated yet. The pattern should be like the other image models, where the image processor is implemented, the feature extractor is an alias and the FE is primarily used (to avoid clashes with docs etc.). I'm currently working on PRs to add deprecation + remove FE references will I'll open soon.

Reply:
OK, but just wondering, the deprecation should only go for existing feature extractors, no? Why we would like to add new feature extractors at this moment and deprecate it later? I mean doing this doesn't seem to bring any value, but you might have some considerations.

Reply:
Agreed. The difficulty is that feature extractors are assumed / expected all over the codebase. Not having one here will likely cause many failures which are beyond this PR and will essentially add a bunch of extra logic to handle both feature extractors and image processors at the same time. They also haven't been officially deprecated, so most people using vision models haven't received warning messages and expect to be able to use `AutoFeatureExtractor` with no problems. As the feature extractor is just an alias for the image processor here, and not its own class, I think it's a pragmatic solution. This should be addressed asap though, as it is confusing.

Reply:
Got it, thank you for the explanation 💯 !